server/models/procedures/migration-process.sql

```diff
@@ -35,17 +35,36 @@ BEGIN
       EXECUTE drop_query;
     END IF;
 END $$
+DELIMITER ;
 
+-- add_column_if_missing:
 
+DROP PROCEDURE IF EXISTS add_column_if_missing;
+
+DELIMITER $$
+CREATE PROCEDURE add_column_if_missing(
+  tname VARCHAR(64),
+  cname VARCHAR(64),
+  typeinfo VARCHAR(128)
+)
+BEGIN
+    IF NOT column_exists(tname, cname)
+    THEN
+      SET @add_column_if_missing = CONCAT('ALTER TABLE `', tname, '` ADD COLUMN `', cname, '` ', typeinfo);
```

Comment:
For some reason, this is resulting in the following error on mysql 5.7.  I believe the default collations changed on MySQL 8 which is why this isn't failing.
```sh
ERROR 1267 (HY000) at line 118: Illegal mix of collations (utf8_general_ci,IMPLICIT) and (utf8_unicode_ci,IMPLICIT) for operation '='
```